step/errors.go

```diff
@@ -39,19 +41,33 @@ func wrapXcodebuildCommandError(cmd Printable, out string, err error) error {
 }
 
 func findXcodebuildErrors(out string) []string {
-	var errors []string
+	var errorLines []string
+	var nserrors []nserror.Error
 
 	scanner := bufio.NewScanner(strings.NewReader(out))
 	scanner.Split(bufio.ScanLines)
 	for scanner.Scan() {
 		line := scanner.Text()
 		if strings.HasPrefix(line, "error: ") {
-			errors = append(errors, line)
+			errorLines = append(errorLines, line)
+		} else if strings.HasPrefix(line, "Error ") {
```

Comment:
This seems like a bit too broad, are you sure it won't catch irrelevant errors accidentally?

Reply:
Good question. What I would say is that this hunt for errors is for a certain xcode command like archive or ipa export. So if a single xcode command writes something to the output prefixed with `Error ` then there is a good chance it could be relevant. 

I think if the final error we print out in the output has more logs then necessary (like we collected some irrelevant errors too) is better then having less information and you need to hunt for it in the build logs. But this is also a bit of experimentation and we will see what the user reaction will be. 😄

Reply:
I was running the step with a couple of projects and tried to reproduce different errors, and both `error: ` and `Error ` prefix search seemed to be fine.

An interesting thing is that the `archive` command didn't seem to produce NSError, only the IPA export command (Obviously, I couldn't check all the possible errors).

Also, if the IPA export produced an error both the `error: ` and NSError (`Error `) are present. The `error: ` seems to be the `NSLocalizedDescription` of its  NSError pair.

An example:

```
error: exportArchive: "code-sign-test.app" requires a provisioning profile.

Error Domain=IDEProvisioningErrorDomain Code=9 ""code-sign-test.app" requires a provisioning profile." UserInfo={IDEDistributionIssueSeverity=3, NSLocalizedDescription="code-sign-test.app" requires a provisioning profile., NSLocalizedRecoverySuggestion=Add a profile to the "provisioningProfiles" dictionary in your Export Options property list.}
```

Reply:
As long as those accidentally matched errors are indeed printed out (even if parsing them as NSError fails), I'm fine with this, just wanted to double-check